athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java

```diff
@@ -206,4 +208,36 @@ public static Object coerceDecimalToExpectedType(BigDecimal value, Types.MinorTy
                 return value;
         }
     }
+
+    public static Object coerceDateTimeToExpectedType(Object value, Types.MinorType fieldType,
+                                                      String customerConfiguredFormat, ZoneId defaultTimeZone)
+    {
+        try {
+            if (value instanceof String) {
+                switch (fieldType) {
+                    case DATEMILLI:
+                        return DateTimeFormatterUtil.stringToLocalDateTime((String) value, customerConfiguredFormat, defaultTimeZone);
+                    case DATEDAY:
+                        return DateTimeFormatterUtil.stringToLocalDate((String) value, customerConfiguredFormat, defaultTimeZone);
+                    default:
+                        return value;
+                }
+            }
+            else if (value instanceof BigDecimal) {
+                switch (fieldType) {
+                    case DATEMILLI:
+                        return DateTimeFormatterUtil.bigDecimalToLocalDateTime((BigDecimal) value, defaultTimeZone);
+                    case DATEDAY:
+                        return DateTimeFormatterUtil.bigDecimalToLocalDate((BigDecimal) value, defaultTimeZone);
+                    default:
+                        return value;
+                }
+            }
+            return value;
+        }
+        catch (IllegalArgumentException ex) {
+            logger.error(ex.getMessage());
```

Comment:
we can throw illegalArgumentexception then and there. it would only throw if any of the date/datetime values fails to parse with any of the custom or supported format. As it is malformed data, we could have the query fail? I continue passing it as string and set vector as string but it would fail on any filtering and such anyways